jdk17-unit-tests/src/test/java/com/uber/nullaway/jdk17/NullAwayInstanceOfBindingTests.java

```diff
@@ -0,0 +1,47 @@
+package com.uber.nullaway.jdk17;
+
+import com.google.errorprone.CompilationTestHelper;
+import com.uber.nullaway.NullAway;
+import java.util.Arrays;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class NullAwayInstanceOfBindingTests {
+
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private CompilationTestHelper defaultCompilationHelper;
+
+  @Before
+  public void setup() {
+    defaultCompilationHelper =
+        CompilationTestHelper.newInstance(NullAway.class, getClass())
+            .setArgs(
+                Arrays.asList(
+                    "-d",
+                    temporaryFolder.getRoot().getAbsolutePath(),
+                    "-XepOpt:NullAway:AnnotatedPackages=com.uber"));
+  }
+
+  @Test
+  public void testInstanceOfBinding() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "InstanceOfBinding.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "class InstanceOfBinding {",
+            "  public void testInstanceOfBinding(@Nullable Object o) {",
+            "    if (o instanceof String s) {",
```

Comment:
Are the semantics here that `null instanceof Foo` is always `false`?

Reply:
Sorry missed this comment, yes that is correct